lib/core/src/Cardano/Wallet/Types/Read/Block.hs

```diff
@@ -0,0 +1,168 @@
+
+{-------------------------------------------------------------------------------
+    Block header
+-------------------------------------------------------------------------------}
+-- TODO:
+-- Define 'getBlockHeight' and 'getSlotNo' in terms of a 'BlockHeader' instead.
+-- For this purpose, remove 'prevBlockHeader' from 'BlockHeader' so that we
+-- do not have to carry the genesis parameters around?
+
+getBlockHeight :: Block -> Quantity "block" Word32
+getBlockHeight = fromBlockNo . \case
+    -- See Note [SeeminglyRedundantPatternMatches]
+    O.BlockByron block ->
+        O.blockNo block
+
+    O.BlockShelley block -> case block of
+        O.ShelleyBlock (Shelley.Block (Shelley.BHeader header _) _) _ ->
+            Shelley.bheaderBlockNo header
+
+    O.BlockAllegra block -> case block of
+        O.ShelleyBlock (Shelley.Block (Shelley.BHeader header _) _) _ ->
+            Shelley.bheaderBlockNo header
+
+    O.BlockMary block -> case block of
+        O.ShelleyBlock (Shelley.Block (Shelley.BHeader header _) _) _ ->
+            Shelley.bheaderBlockNo header
+
+    O.BlockAlonzo block -> case block of
+        O.ShelleyBlock (Shelley.Block (Shelley.BHeader header _) _) _ ->
+            Shelley.bheaderBlockNo header
+
+    O.BlockBabbage block -> case block of
+        O.ShelleyBlock (Shelley.Block (O.Header header _) _) _ ->
+            O.hbBlockNo header
+
+-- FIXME unsafe conversion (Word64 -> Word32)
+fromBlockNo :: O.BlockNo -> Quantity "block" Word32
+fromBlockNo (O.BlockNo h) = Quantity (fromIntegral h)
+
+getSlotNo :: Block -> O.SlotNo
+getSlotNo = \case
+    -- See Note [SeeminglyRedundantPatternMatches]
+    O.BlockByron block ->
+        O.blockSlot block
+
+    O.BlockShelley block -> case block of
+        O.ShelleyBlock (Shelley.Block (Shelley.BHeader header _) _) _ ->
+            Shelley.bheaderSlotNo header
+
+    O.BlockAllegra block -> case block of
+        O.ShelleyBlock (Shelley.Block (Shelley.BHeader header _) _) _ ->
+            Shelley.bheaderSlotNo header
+
+    O.BlockMary block -> case block of
+        O.ShelleyBlock (Shelley.Block (Shelley.BHeader header _) _) _ ->
+            Shelley.bheaderSlotNo header
+
+    O.BlockAlonzo block -> case block of
+        O.ShelleyBlock (Shelley.Block (Shelley.BHeader header _) _) _ ->
+            Shelley.bheaderSlotNo header
+
+    O.BlockBabbage block -> case block of
+        O.ShelleyBlock (Shelley.Block (O.Header header _) _) _ ->
+            O.hbSlotNo header
+
+{-------------------------------------------------------------------------------
+    Block transactions
+-------------------------------------------------------------------------------}
+-- | Retrieve the sequence of 'Tx' contained in a block, as a list.
+getTxsListFromBlock :: Block -> [Read.Tx]
+getTxsListFromBlock = toList . getTxsFromBlock
+
+-- | Count the number of transactions in a 'Block'.
+numberOfTransactionsInBlock :: Block -> Int
+numberOfTransactionsInBlock = Seq.length . getTxsFromBlock
```

Comment:
Given that you've already used `Txs` elsewhere, would suggest to keep the same convention here:

```suggestion
numberOfTxsInBlock :: Block -> Int
numberOfTxsInBlock = Seq.length . getTxsFromBlock
```